app/ui/browser-modern/views/browser/page/index.jsx

```diff
@@ -73,33 +80,75 @@ class Page extends Component {
       this.props.dispatch(UIEffects.focusURLBar(this.props.pageId, { select: isTofino }));
     });
 
+    this.webview.addEventListener('did-finish-loading', () => {
+      // Event fired only when page is successfully loaded
+      this.props.dispatch(PageActions.setPageState(this.props.pageId, {
+        load: PageState.STATES.LOADED,
+      }));
+    });
+
+    this.webview.addEventListener('did-stop-loading', () => {
+      // Event fired when a page finishes loading.
+      // May be emitted multiple times for every single frame.
+      // Emitted regardless of whether or not a failure occurs.
+      const url = this.webview.getURL();
+      const title = this.webview.getTitle();
+
+      this.props.dispatch(PageActions.setPageDetails(this.props.pageId, {
+        canGoBack: this.webview.canGoBack(),
+        canGoForward: this.webview.canGoForward(),
+        location: url,
+        title,
```

Comment:
Can the title ever be different here? Why update it when we have the `page-title-set` event?